README.md

```diff
@@ -18,5 +19,60 @@ Clone the repo using:
 `git clone https://github.com/DIT168-V2V-responsibles/v2v-protocol.git`
 
 ### 3. License
+The protocol is licenced under GNU Lesser General Public License version 3.0. This is due to the incorporation of "libcluon" library as part of the project. Libcluon offers their software under LGPLv 3.0 licence and due to the copyleft nature, anyone who distribute its code or derivative works, are required to make the source available under the same terms. 
+Libcluon library can be found at https://github.com/chrberger/libcluon
 
-This repo uses the license [GPLv3](https://www.gnu.org/licenses/gpl-3.0.en.html). Please refer to the official documentation before you contribute.
+### 4. Protocol Requests 
+This section describes the the the protocol requests. Fields of requests and their types are denoted: Type field name.
+
+#### 4.1 Common Requests
+
+##### Announce Presence
+This message is intended as a way of letting cars in the network know about each other. Each car should sent an Announce Presence when joining a platoon. Different channels are used to distinguish between different platoons. 
+
+***Fields***
+* unit_8_t channel - A unique identifier of the car sending the follow announcement.
+
+##### Follow Request  
+This message is sent to a channel when a car is attempting to initiate following. This message requires a response i.e. Follow Response. 
+
+***Fields***
+* unit_8_t carId - A unique identifier of the car that initiated the following.
+
+##### Follow Response
+This message is sent in response to a Follow Request. The message returns the identifier of the car at the last position of the platoon and the car ID that originally initiated the following.
+
+***Fields***
+* unit_8_t meantForCar - A unique identifier of the car that initiated the following.
+* unit_8_t carToFollow - A unique identifier of the final car within the platoon line.
+
+##### Emergency Brake
+This message is sent in order to stop the cars from moving. The request is used to avoid any collision and possible damage between the cars. 
+
```

Comment:
State if the message expects a response or not!

Reply:
added